### ISSUE TITLE
lint against shadowing prelude types in the signature of public functions

### ISSUE THREAD

**Issue by lolbinarycat** [rust-lang/rust-clippy#13426]
### What it does

warn if any item that shares a name with an item in the prelude is used without qualification in the signature of a public function.

### Advantage

people might see the auto-generated documentation and think it is referring to the prelude type instead of another type.

### Drawbacks

this pattern is extremely common in the rust ecosystem, especially with `Result` types.

### Example

```rust
use std::io::Result;

fn unclear() -> Result<()> {
    Ok(())
}
```

Could be written as:

```rust
use std::io;

fn unclear() -> io::Result<()> {
    Ok(())
}
```

**Comment by PatchMixolydic** [comment]
rustdoc still renders `io::Result` as `Result`:
![Image](https://github.com/user-attachments/assets/282f6c3e-627d-4e51-a597-11461e0b624e)

The real fix would be something like:
```rust
use std::io::Result as IoResult;

fn unclear() -> IoResult<()> {
    Ok(())
}
```

Or even better:
```rust
use std::io::Error as IoError;

fn unclear() -> Result<(), IoError> {
    Ok(())
}
```

In my completely useless opinion, naming public types `Error` or creating type aliases for `Result` should be considered an antipattern for exactly this reason. (`Result` aliases have the added downsides of obfuscating what the `E` type is and making it difficult to determine whether the alias is related to `core::result::Result` or not.)
